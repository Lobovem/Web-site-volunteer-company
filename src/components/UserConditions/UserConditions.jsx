import s from './UserConditions.module.scss';

export const UserConditions = () => {
  return (
    <div className={s.userCondition}>
      <h3 className={s.userCondition__decor}>Угода</h3>

      <p className={s.userCondition__desc}>
        Я («Користувач») погоджуюсь з умовами цієї Угоди користувача і, використовуючи сайт: ukrhumbat.com.ua (надалі – «Сайт»), у разі направлення заявки на
        отримання допомоги, надаю згоду на обробку і захист своїх персональних даних, як Користувача, в обсязі, порядку, відповідно до мети обробки.
      </p>

      <h2 className={s.userCondition__title}>Персональні дані</h2>
      <p className={s.userCondition__desc}>
        На виконання вимог Закону України "Про захист персональних даних" шляхом заповнення заявки на отримання допомоги даю згоду на обробку вказаних у заявці
        моїх персональних даних.
      </p>
    </div>
  );
};
