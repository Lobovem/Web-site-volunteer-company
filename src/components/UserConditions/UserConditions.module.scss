@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.userCondition {

  position: relative;

  @media (min-width: 1500px) {
    padding-bottom: 150px;
  }



  &__decor {
    margin: 0;
    padding: 0;
    display: none;

    @media (min-width: 1500px) {

      position: absolute;
      display: block;
      top: 50px;
      left: -370px;
      z-index: 0;
      font-weight: $font-weight-500;
      font-size: $font-size-165;
      line-height: 165px;
      text-align: center;
      text-transform: uppercase;
      color: $color-grey;
      mix-blend-mode: normal;
      transform: rotate(-90deg);
    }
  }

  &__title {
    font-weight: $font-weight-500;
    font-size: $font-size-18;
    line-height: 18px;
    color: $color-black;
    mix-blend-mode: normal;
    margin: 25px 0;

    @include lg-min {
      font-weight: $font-weight-500;
      font-size: $font-size-24;
      line-height: 24px;
    }
  }

  &__desc {
    font-weight: $font-weight-400;
    font-size: $font-size-14;
    line-height: 21px;
    color: $color-black;
    mix-blend-mode: normal;
    margin-bottom: 10px;
  }
}