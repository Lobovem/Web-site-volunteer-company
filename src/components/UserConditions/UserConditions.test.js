import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { UserConditions } from './UserConditions';

describe('UserConditions', () => {
  test('renders UserConditions component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(<UserConditions />);
    expect(component).toMatchSnapshot();
  });

  test('search tag h3 decor', () => {
    render(<UserConditions />);
    const h3Element = screen.getByRole('heading', { level: 3 });
    expect(h3Element).toBeInTheDocument();
  });

  test('search tag h2 title', () => {
    render(<UserConditions />);
    const h2Element = screen.getByRole('heading', { level: 2 });
    expect(h2Element).toBeInTheDocument();
  });
});
