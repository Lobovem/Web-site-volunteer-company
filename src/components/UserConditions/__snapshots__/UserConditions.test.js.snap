// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserConditions renders UserConditions component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="userCondition"
      >
        <h3
          class="userCondition__decor"
        >
          Угода
        </h3>
        <p
          class="userCondition__desc"
        >
          Я («Користувач») погоджуюсь з умовами цієї Угоди користувача і, використовуючи сайт: ukrhumbat.com.ua (надалі – «Сайт»), у разі направлення заявки на отримання допомоги, надаю згоду на обробку і захист своїх персональних даних, як Користувача, в обсязі, порядку, відповідно до мети обробки.
        </p>
        <h2
          class="userCondition__title"
        >
          Персональні дані
        </h2>
        <p
          class="userCondition__desc"
        >
          На виконання вимог Закону України "Про захист персональних даних" шляхом заповнення заявки на отримання допомоги даю згоду на обробку вказаних у заявці моїх персональних даних.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="userCondition"
    >
      <h3
        class="userCondition__decor"
      >
        Угода
      </h3>
      <p
        class="userCondition__desc"
      >
        Я («Користувач») погоджуюсь з умовами цієї Угоди користувача і, використовуючи сайт: ukrhumbat.com.ua (надалі – «Сайт»), у разі направлення заявки на отримання допомоги, надаю згоду на обробку і захист своїх персональних даних, як Користувача, в обсязі, порядку, відповідно до мети обробки.
      </p>
      <h2
        class="userCondition__title"
      >
        Персональні дані
      </h2>
      <p
        class="userCondition__desc"
      >
        На виконання вимог Закону України "Про захист персональних даних" шляхом заповнення заявки на отримання допомоги даю згоду на обробку вказаних у заявці моїх персональних даних.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
